corehq/apps/users/audit/change_messages.py

```diff
@@ -1,91 +1,294 @@
+    ),
+    "reset_password": noop("Password reset"),
+    "activate_user": noop('User re-enabled. Reason: "{reason}"'),
+    "deactivate_user": noop('User disabled. Reason: "{reason}"'),
+    "add_phone_numbers": noop('Added phone number(s) {phone_numbers}'),
+    "remove_phone_numbers": noop('Removed phone number(s) {phone_numbers}'),
+    "set_profile": noop("Profile: {name}[{id}]"),
+    "clear_profile": noop("Profile: None"),
+    "set_primary_location": noop("Primary location: {name}[{id}]"),
+    "clear_primary_location": noop("Primary location: None"),
+    "set_assigned_locations": noop("Assigned locations: {locations_info}"),
+    "clear_assigned_locations": noop("Assigned locations: []"),
+    "set_groups": noop("Groups: {groups_info}"),
+    "clear_groups": noop("Groups: []"),
+    "add_domain_invitation": noop("Invited to domain '{domain}'"),
+    "remove_domain_invitation": noop("Invitation revoked for domain '{domain}'")
+}
+
+
+class UserChangeMessageV1(object):
+    """
+    Each change message to follow the structure
+    {
+        "field": {
+            "slug": message_slug,
+            "params": {} # optional params for the message
+        }
+    }
+    field: could be domain, phone_numbers etc
+    it is needed as the top-level key to make it searchable
+    slug: any slug in MESSAGES
+    params: (optional) params for the message
+    """
     @staticmethod
     def program_change(program):
         if program:
-            message = f"Program: {program.name}[{program.get_id}]"
+            change_message = {"program": [
+                {
+                    "slug": "set_program",
+                    "params": {"id": program.get_id, "name": program.name}
+                }
+            ]}
         else:
-            message = 'Program: None'
-        return message
+            change_message = {"program": [
+                {"slug": "clear_program"}
+            ]}
+        return change_message
 
     @staticmethod
     def role_change(user_role):
-        message = 'Role: None'
         if user_role:
-            message = f"Role: {user_role.name}[{user_role.get_qualified_id()}]"
-        return message
+            change_message = {"role": [
+                {
+                    "slug": "set_role",
+                    "params": {"id": user_role.get_qualified_id(), "name": user_role.name}
+                }
+            ]}
+        else:
+            change_message = {"role": [
+                {"slug": "clear_role"}
+            ]}
+        return change_message
 
     @staticmethod
     def domain_removal(domain):
-        return f"Removed from domain '{domain}'"
+        return {"domain": [
+            {
+                "slug": "remove_from_domain",
+                "params": {"name": domain}
+            }
+        ]}
 
     @staticmethod
     def registered_devices_reset():
-        return "Registered devices reset"
+        return {"devices": [
+            {
+                "slug": "reset_devices"
```

Comment:
as mentioned before I don't think we need this at all and should combine the 3 'two_factor' events into one